### DISCUSSION TITLE
A beginner getting an error when trying to push

### DISCUSSION THREAD

**Discussion by Pitxunet** [Syknapse/Contribute-To-This-Project#1757] [Q&A] [answered]
Hi,
I'm sorry if it's not the right place to post this, but I'm getting an error when trying to push my branch to the original project. 
I wrote: git push origin -u add-Estitxu-card
And I'm getting:
remote: Permission to Syknapse/Contribute-To-This-Project.git denied to Pitxunet.
fatal: unable to access 'https://github.com/Syknapse/Contribute-To-This-Project.git/': The requested URL returned error: 403

I understand that I get to the webpage but I'm unable to access.. I don't understand why. I participated in another similar beginner project two days ago and it worked like a charm...
Could anyone please give me a hand or some clue as to why this is happening? Thank you!
Estitxu

**Comment by Pitxunet** [comment]
Well, I figured it out. There was a password issue so first I created a token (Thank you stack over flowers!). But then the fork kept getting behind before I had the time to push my version... Not sure if there is a solution for this or if I just need to wait for the repository activity to slow down?

**Comment by ChrisGk89** [comment, marked as answer]
Since I didn't understood if your problem is solved but I want to help anyway. If your push didn't accepted there are many issues that you should check. 

**1. It is the password issue as you mention and you solved it creating a token.** 

**3. You should check the permissions of the branch.**

**4. If your branch is behind the current branch check this solution: (assuming your local is in sync with origin i.e. you've run git fetch already):**

  `$ git checkout mybranch`

 `$ git branch --set-upstream-to=origin/mybranch mybranch`

 ` $ git merge main`

  `$ git push origin mybranch`

Another approach is: 

  `$ git checkout main` (you are switching your branch to main)

  `$ git pull `

  `$ git checkout yourBranch` (switch back to your branch)

  `$ git merge main`

Check this site for more: [https://9to5answer.com/branch-is-behind-main](url)

**5. Other solution is to rebase.** 

  `$ git checkout yourbrach`

  `$ git rebase master`

  First, rewinding head to replay your work on top of it...
  Applying: added staged command 

  `$ git checkout master`

  `$ git merge experiment`

  You can check for more here: [https://git-scm.com/book/en/v2/Git-Branching-Rebasing](url) 

**6. Another solution is Stash**

 This way you take completely the changes and clear the directory. When you are in the middle of something, you learn that there are upstream changes that are possibly relevant to what you are doing. When your local changes do not conflict with the changes in the upstream, a simple git pull will let you move forward.

However, there are cases in which your local changes do conflict with the upstream changes, and git pull refuses to overwrite your changes. In such a case, you can stash your changes away, perform a pull, and then unstash, like this: 
You should follow the commands below:

  `$ git pull`
 ...
file foobar not up to date, cannot merge.
  `$ git stash`

  `$ git pull`

  `$ git stash pop`

You can find all the information about stash here: [https://git-scm.com/docs/git-stash](url)

Hope I help you in any way.

Best wishes for your future projects.

**Reply by Pitxunet** [reply]
Thank you. What I did in the end was to try to push my changes faster so there were no other conflicting pushes in the meanwhile but I think stashing sounds just right. I will try to remember next time I find a similar issue.

**Comment by zeon-X** [comment]
If your are in VS code. Try to use Source Control (located at the left corner) It will help you sign-in and other staff easily

**Comment by visionthex** [comment]
For beginners they can use the GitHub desktop.  From others that have used it its easier than the command line.
https://desktop.github.com/

**Comment by kecalek13** [comment]
Because you are trying to push it to this repository you need to fork this repository and push it to your forked repository and then merge your forked repository to original (this)

**Reply by Gmoneyshungry47** [reply]
Ty im fullyninterestedin keeping everyone happily connected
